canopen/network.py

```diff
@@ -279,6 +279,21 @@ def __len__(self) -> int:
         return len(self.nodes)
 
 
+class _UninitializedNetwork(Network):
+    """Empty network implementation as a placeholder before actual initialization."""
+
+    def __init__(self, bus: Optional[can.BusABC] = None):
+        """Do not initialize attributes, by skipping the parent constructor."""
+
+    def __getattribute__(self, name):
+        raise RuntimeError("No actual Network object was assigned, "
+                           "try associating to a real network first.")
+
+
+#: Singleton instance
+_UNINITIALIZED_NETWORK = _UninitializedNetwork()
```

Comment:
```suggestion
_UNINITIALIZED_NETWORK: Final[Network] = _UninitializedNetwork()
```

Requires `from typing import Final`.